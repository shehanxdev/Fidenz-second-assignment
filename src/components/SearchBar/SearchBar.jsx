import React, { useState } from "react";
import { CARD_COLORS } from "../../constants";

export function SearchBar() {
  const [searchTerm, setSearchTerm] = useState("");

  const handleSearchChange = (event) => {
    setSearchTerm(event.target.value);
  };

  const handleAddButtonClick = () => {
    alert("This feature is under construction");
  };

  return (
    <div className="d-flex justify-content-center ">
      <input
        className="border-0  p-2 rounded-2 text-light"
        type="text"
        placeholder=" Search..."
        value={searchTerm}
        onChange={handleSearchChange}
        style={{
          backgroundColor: CARD_COLORS.black,
          height: "2.5rem",
          width: "18rem",
        }}
      />
      <button
        className="px-5 py-2 rounded "
        onClick={handleAddButtonClick}
        style={{
          marginLeft: "-8px",

          backgroundColor: CARD_COLORS.purple,
          color: "white",
          border: "none",
          cursor: "pointer",
        }}
      >
        Add
      </button>
    </div>
  );
}

export default SearchBar;
