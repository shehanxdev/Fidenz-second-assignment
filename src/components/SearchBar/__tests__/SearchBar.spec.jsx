import React from "react";
import renderer from "react-test-renderer";
import { SearchBar } from "./../SearchBar";

describe("Header component", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<SearchBar />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
