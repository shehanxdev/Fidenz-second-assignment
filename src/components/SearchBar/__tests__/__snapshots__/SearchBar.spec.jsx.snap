// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders correctly 1`] = `
<div
  className="d-flex justify-content-center "
>
  <input
    className="border-0  p-2 rounded-2 text-light"
    onChange={[Function]}
    placeholder=" Search..."
    style={
      Object {
        "backgroundColor": "#1F2128",
        "height": "2.5rem",
        "width": "18rem",
      }
    }
    type="text"
    value=""
  />
  <button
    className="px-5 py-2 rounded "
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#6249CC",
        "border": "none",
        "color": "white",
        "cursor": "pointer",
        "marginLeft": "-8px",
      }
    }
  >
    Add
  </button>
</div>
`;
